Sources/Swiftly/SelfUpdate.swift

```diff
@@ -31,50 +39,76 @@ struct SelfUpdate: SwiftlyCommand {
             )
         }
 
-        let _ = try await Self.execute(ctx, verbose: self.root.verbose)
+        let _ = try await Self.execute(ctx, verbose: self.root.verbose, version: self.toVersion)
     }
 
-    public static func execute(_ ctx: SwiftlyCoreContext, verbose: Bool) async throws
+    public static func execute(_ ctx: SwiftlyCoreContext, verbose: Bool, version swiftlyVersion: SwiftlyVersion?) async throws
         -> SwiftlyVersion
     {
-        await ctx.print("Checking for swiftly updates...")
+        var downloadURL: Foundation.URL?
+        var version: SwiftlyVersion? = swiftlyVersion
 
-        let swiftlyRelease = try await ctx.httpClient.getCurrentSwiftlyRelease()
+        if let version {
+#if os(macOS)
+            downloadURL = URL(string: "https://download.swift.org/swiftly/darwin/swiftly-\(version).pkg")
+#endif
 
-        guard try swiftlyRelease.swiftlyVersion > SwiftlyCore.version else {
-            await ctx.print("Already up to date.")
-            return SwiftlyCore.version
+#if os(Linux)
+#if arch(x86_64)
+            downloadURL = URL(string: "https://download.swift.org/swiftly/linux/swiftly-\(version)-x86_64.tar.gz")
+#elseif arch(arm64)
+            downloadURL = URL(string: "https://download.swift.org/swiftly/linux/swiftly-\(version)-aarch64.tar.gz")
+#endif
```

Comment:
Should this throw an error if we somehow get here and don't know what we're downloading?
As it stands, I think this will end up running the update based on `getCurrentSwiftlyRelease()` since `downloadURL` won't get set, even though a version was specified.